trunk/NDHMS/Routing/module_channel_routing.F

```diff
@@ -373,20 +432,41 @@ subroutine SUBMUSKINGCUNGE(    &
             if(D .eq. 0.0) then
               print *, "FATAL ERROR: D is 0 in MUSKINGCUNGE", Km, X, dt,D
               call hydro_stop("In MUSKINGCUNGE() - D is 0.")
+! More old code. I kept it here as a reference
+!           endif 
+  
+!           C1 =  (Km*X + dt/2)/D
+!           C2 =  (dt/2 - Km*X)/D
+!           C3 =  (Km*(1-X)-dt/2)/D
+!           C4 =  (ql*dt)/D                       !-- ql already multipled by the dx length
+!           C4 =  (ql*dt)/D  - (ChannK * dx * WPk)  !-- (loss units: m/s * m * m)
+           !TML-- channel loss now uses modified WP (from KINEROS2)
+
+!           if(R .le. 0.0) then 
+!             Qj =  ((C1*qup)+(C2*quc)+(C3*qdp) + C4)
+!           else
+!              if( (C4 .lt. 0.0) .and. (abs(C4) .gt. (C1*qup)+(C2*quc)+(C3*qdp)))  then
+!               C4 = -((C1*qup)+(C2*quc)+(C3*qdp))
+!              endif
+!             Qj =  ((C1*qup)+(C2*quc)+(C3*qdp) + C4) -((1/n) * AREA * (R**(2./3.)) * sqrt(So))
+! End of old code, GitHub moved the endif statements around in a strange way, feel free to move.
            endif
 
            C1 =  (Km*X + dt/2.000000)/D
            C2 =  (dt/2.000000 - Km*X)/D
            C3 =  (Km*(1.000000-X)-dt/2.000000)/D
 !          C1 =  max(0.0,min(1.0,1.0-C3)) !! eliminate influence of upstream current
-           C4 =  (ql*dt)/D
-!          C4 =  (ql*dt)/D  - (ChannK * dx * (WP+WPC))  !-- (loss units: m/s * m * m)
+!           C4 =  (ql*dt)/D
+           C4 =  (ql*dt)/D  - (ChannK * dx * (WP+WPC))  !-- (loss units: m/s * m * m)
 
            if( (C4 .lt. 0.0) .and. (abs(C4) .gt. (C1*qup)+(C2*quc)+(C3*qdp)))  then
             C4 = -((C1*qup)+(C2*quc)+(C3*qdp))
            endif
 
            if((WP+WPC) .gt. 0.0) then
+            if( (C4 .lt. 0.0) .and. (abs(C4) .gt. (C1*qup)+(C2*quc)+(C3*qdp)))  then
```

Comment:
Channel infiltration in the upper interval should be the same as the lower interval. Is this logic preventing negative flow still ok?